backend/common/ai_drive.py

```diff
@@ -145,29 +144,39 @@ def dl_tabular_drive(
     return train_loss_results
 
 
-def dl_img_drive(
-    train_transform,
-    test_transform,
-    user_arch,
-    criterion,
-    optimizer_name,
-    default,
-    epochs,
-    batch_size,
-    shuffle,
-    IMAGE_UPLOAD_FOLDER,
-):
-    print(user_arch)
+def dl_img_drive(trainspace_data: TrainspaceData):
+    params = trainspace_data.parameters_data
+
+    optimizer_name = params["optimizer_name"]
+    criterion = params["criterion"]
+    default = (
+        trainspace_data.dataset_data["name"]
+        if trainspace_data.dataset_data.get("is_default_dataset")
+        else None
+    )
+    epochs = params.get("epochs", 5)
+    shuffle = params.get("shuffle", True)
+    batch_size = params.get("batch_size", 20)
+    train_transform = params["train_transform"]
+    test_transform = params["test_transform"]
+
+    user_arch = params["layers"]
+
     model = DLModel(parse_deep_user_architecture(user_arch))
```

Comment:
next step would be to refactor the `parse_deep_user_architecture` to be compatible with the `params["layers"]` from `trainspace`